### PR TITLE
[Fix:#2264] Adding OutputFormatters to resource filters 

### PR DESCRIPTION
**Pull request by harshgMSFT**

### CODE REVIEW

File: src/Microsoft.AspNet.Mvc.Core/ActionResults/ObjectResult.cs [hunk 1]
```diff
@@ -273,10 +273,14 @@ private IEnumerable<IOutputFormatter> GetDefaultFormatters(ActionContext context
...
+                var actionBindingContext = context.HttpContext
                                     .RequestServices
-                                    .GetRequiredService<IOutputFormattersProvider>()
-                                    .OutputFormatters;
+                                    .GetRequiredService<IScopedInstance<ActionBindingContext>>()
+                                    .Value;
+
+                // In scenarios where there is a resource filter which directly shortcircuits using an ObjectResult.
+                // actionBindingContext is not setup yet and is null.
+                formatters = actionBindingContext?.OutputFormatters ?? new List<IOutputFormatter>();
```
Comment (rynowak) [inline src/Microsoft.AspNet.Mvc.Core/ActionResults/ObjectResult.cs]: Should we fall back to options in this case? Seems better than always giving a 406.

**Comment by rynowak** [comment]
:watch: code seems fine, this will have to get rebased on top of pranav's changes of course.

Let's discuss the scenario where a resource filter short-circuits.

File: src/Microsoft.AspNet.Mvc.Core/ActionResults/ObjectResult.cs (hunk 1 above)
Comment (harshgMSFT) [inline src/Microsoft.AspNet.Mvc.Core/ActionResults/ObjectResult.cs]: sure that seems fine, however that would mean another call to `GetRequiredService`

**Comment by harshgMSFT** [comment]
@rynowak updated

File: src/Microsoft.AspNet.Mvc.Core/ControllerActionInvoker.cs [hunk 2]
```diff
@@ -31,9 +32,10 @@ public class ControllerActionInvoker : FilterActionInvoker
             [NotNull] IScopedInstance<ActionBindingContext> actionBindingContextAccessor,
             [NotNull] ITempDataDictionary tempData)
             : base(
-                  actionContext, 
+                  actionContext,
                   filterProviders,
-                  inputFormatters, 
+                  inputFormatters,
+                  outputFormatters,
```
Comment (rynowak) [inline src/Microsoft.AspNet.Mvc.Core/ControllerActionInvoker.cs]: Just as an FYI - this will break entropy, and you will have to update it.

**Comment by rynowak** [comment]
:shipit:

**Comment by harshgMSFT** [comment]
58a5ad22797f5dc3589a341832715dc188b0b3e9

**Pull request closed without merging**